src/app/editor/abstract-array-field/abstract-array-field.component.ts

```diff
@@ -0,0 +1,66 @@
+import { AbstractTrackerComponent } from '../abstract-tracker';
+
+import { EmptyValueService } from '../shared/services';
+
+/**
+ * Abstract component to share code of common operations of all array fields
+ * 
+ * Instance properties declared here only to resolve syntax errors. 
+ * Hence they need to be declared in children extending components (with decarators if necessary)
+ */
+export abstract class AbstractArrayFieldComponent extends AbstractTrackerComponent {
+
+  values: Array<any>;
+  schema: Object;
+  emptyValueService: EmptyValueService;
+  protected _emptyValue: any;
+
+  /**
+   * Called when a property of any element of the values is changed
+   * Used if values is a object array.
```

Comment:
@jmartinm 
Actually it is not. It is used all array fields except the `PrimitiveArrayFieldComponent`. This `ObjectArrayFieldComponent` is wrongly named, we should change it to `TableArrayFieldComponent` or something that has "table" in it.